modules/distribution/resources/sql/appmgt/oracle.sql

```diff
@@ -408,13 +408,17 @@ CREATE OR REPLACE TRIGGER APM_API_CONSUMER_APPS_TRIGGER
 
 
 
--- TODO : Add Foreign key contraints for UUID --> APM_APP(UUID) and USER_ID --> APM_SUBSCRIBER(USER_ID),
-CREATE TABLE APM_APP_HIT_TOTAL (
+CREATE TABLE IF NOT EXISTS APM_APP_HITS (
     UUID VARCHAR2(500) NOT NULL,
+    APP_NAME VARCHAR2(200) NOT NULL,
+    VERSION VARCHAR2(50),
+    CONTEXT VARCHAR2(256) NOT NULL,
     USER_ID VARCHAR2(50) NOT NULL,
-    HIT_COUNT NUMBER(19),
     TENANT_ID INTEGER,
-    PRIMARY KEY (UUID, USER_ID)
+    HIT_TIME DATE NOT NULL,
+    PRIMARY KEY (UUID, USER_ID, TENANT_ID, HIT_TIME),
+    FOREIGN KEY (UUID) REFERENCES APM_APP(UUID) ON UPDATE CASCADE ON DELETE CASCADE,
```

Comment:
Better not to use cascade on delete in summery table as not to loose the history data. We will address this in a seperate jira as we need to to modify the graphs accordingly
[https://wso2.org/jira/browse/APPM-1591]